.github/workflows/check-state-compatibility.yml

```diff
@@ -0,0 +1,158 @@
+#
+# DO NOT TRIGGER THIS WORKFLOW ON PUBLIC FORKS
+#
+# This workflow runs on a self-hosted runner and forks to this repository
+# can potentially run dangerous code on the self-hosted runner machine
+# by creating a pull request that executes the code in a workflow.
+#
+# ******************************************************************************
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - "*"
+
+env:
+  GOLANG_VERSION: 1.18
+  GENESIS_URL: https://github.com/osmosis-labs/networks/raw/main/osmosis-1/genesis.json
+  ADDRBOOK_URL: https://dl2.quicksync.io/json/addrbook.osmosis.json
+  SNAPSHOT_BUCKET: https://snapshots.osmosis.zone
+  RPC_ENDPOINT: https://rpc.osmosis.zone
+  LCD_ENDPOINT: https://lcd.osmosis.zone
+  DELTA_HALT_HEIGHT: 50
+
+jobs:
+  compare_versions:
+    # Compare current mainnet osmosis major version with the major version of github branch
+    # Skip the next job if the current github major is greater than the current mainnet major
+    runs-on: self-hosted
+    outputs:
+      mainnet_major_version: ${{ steps.mainnet_version.outputs.mainnet_major_version }}
+    steps:
+      - name: Get mainnet major version
+        id: mainnet_version
+        run: |
+          # Find current major version via rpc.osmosis.zone/abci_info
+          RPC_ABCI_INFO=$(curl -s --retry 5 --retry-delay 5 --connect-timeout 30 -H "Accept: application/json" ${{ env.RPC_ENDPOINT }}/abci_info)
+          MAINNET_MAJOR_VERSION=$(echo $RPC_ABCI_INFO | dasel --plain -r json  'result.response.version' | cut -f 1 -d '.')
+
+          echo "MAINNET_MAJOR_VERSION=$MAINNET_MAJOR_VERSION" >> $GITHUB_ENV
+          echo "mainnet_major_version=$MAINNET_MAJOR_VERSION" >> $GITHUB_OUTPUT
+
+  check_state_compatibility:
+    # DO NOT CHANGE THIS: please read the note above
+    needs: compare_versions
+    runs-on: self-hosted
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          repository: osmosis-labs/osmosis
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: 🐿 Setup Golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: "^${{ env.GOLANG_VERSION }}"
+      - name: 🔨 Build the osmosisd binary
+        run: |
+          make build
+          build/osmosisd version
+      - name: 🧪 Initialize Osmosis Node
+        run: |
+          rm -rf $HOME/.osmosisd/ || true
+          build/osmosisd init runner -o
+
+          # Download genesis file if not present
+          mkdir -p /mnt/data/genesis/osmosis-1/
+          if [ ! -f "/mnt/data/genesis/osmosis-1/genesis.json" ]; then
+            wget -q -O /mnt/data/genesis/osmosis-1/genesis.json ${{ env.GENESIS_URL }}
+          fi
+
+          # Copy genesis to config folder
+          cp /mnt/data/genesis/osmosis-1/genesis.json $HOME/.osmosisd/config/genesis.json
+      - name: ⏬ Download last pre-epoch snapshot
+        run: |
+          REPO_MAJOR_VERSION=v14
```

Comment:
had to hardcode this value because currently, our `github describe --tags` out puts this; 
```
❯ git describe --tags
osmomath/v0.0.2-20230103003823-2ac5d3561928-152-gb58890871
```